// @flowdegree/arabic-strings/dist/index.mjs
// src/index.ts
var TASHKEEL = [
  "\u0610",
  // ( ؐ) arabic sign sallallahou alayhe wassallam
  "\u0611",
  // ( ؑ) arabic sign alayhe assallam
  "\u0612",
  // ( ؒ) arabic sign rahmatullah alayhe
  "\u0613",
  // ( ؓ) arabic sign radi allahou anhu
  "\u0614",
  // ( ؔ) arabic sign takhallus
  "\u0615",
  // ( ؕ) arabic small high tah
  "\u0616",
  // ( ؖ) arabic small high ligature alef with lam with yeh
  "\u0617",
  // ( ؗ) arabic small high zain
  "\u0618",
  // ( ؘ) arabic small fatha
  "\u0619",
  // ( ؙ) arabic small damma
  "\u061A",
  // ( ؚ) arabic small kasra
  "\u064B",
  // ( ً) arabic fathatan
  "\u064C",
  // ( ٌ) arabic dammatan
  "\u064D",
  // ( ٍ) arabic kasratan
  "\u064E",
  // ( َ) arabic fatha
  "\u064F",
  // ( ُ) arabic damma
  "\u0650",
  // ( ِ) arabic kasra
  "\u0651",
  // ( ّ) arabic shadda
  "\u0652",
  // ( ْ) arabic sukun
  "\u0653",
  // ( ٓ) arabic maddah above
  "\u0654",
  // ( ٔ) arabic hamza above
  "\u0655",
  // ( ٕ) arabic hamza below
  "\u0656",
  // ( ٖ) arabic subscript alef
  "\u0657",
  // ( ٗ) arabic inverted damma
  "\u0658",
  // ( ٘) arabic mark noon ghunna
  "\u065A",
  // ( ٚ) arabic vowel sign small v above
  "\u065B",
  // ( ٛ) arabic vowel sign inverted small v above
  "\u065C",
  // ( ٜ) arabic vowel sign dot below
  "\u065D",
  // ( ٝ) arabic reversed damma
  "\u065E"
  // ( ٞ) arabic fatha with two dots
];
var ALIF_REGEX = new RegExp("[\u0623\u0625\u0622]", "g");
var YA_REGEX = new RegExp("[\u0649\u064A]", "g");
var TA_REGEX = new RegExp("[\u0629]", "g");
var TASHKEEL_REGEX = new RegExp(TASHKEEL.join(""), "g");
var TATWEEL_REGEX = new RegExp("\u0640", "g");
function howArabic(str) {
  str = str.replace(/[\u0021-\u0040\s]/gm, "");
  const match = str.match(/[\u0621-\u0652]/gm) || [];
  const result = match.length / str.length;
  return result;
}
function howNotArabic(str) {
  str = str.replace(/[\u0021-\u0040\s]/gm, "");
  const match = str.match(/[^\u0621-\u0652]/gm) || [];
  const result = match.length / str.length;
  return result;
}
function isArabic(str, threshold = 0.79) {
  return howArabic(str) >= threshold;
}
function hasArabic(str) {
  return /[\u0621-\u064A]/.test(str);
}
function removeTashkel(str) {
  const escapedTashkeel = TASHKEEL.map((t) => "\\" + t).join("|");
  const regex = new RegExp(escapedTashkeel, "g");
  return str.replace(regex, "");
  return str.replace(TASHKEEL_REGEX, "");
}
function removeNonArabic(str, excludeChars = "") {
  const exclusionRegex = new RegExp(`[${excludeChars}]`, "gm");
  return str.replace(new RegExp(`[^\\u0621-\\u0652${excludeChars}]`, "gm"), "");
}
function removeArabic(str) {
  return str.replace(/[\u0621-\u0652]/gm, "");
}
function removeTatwel(str) {
  return str.replace(TATWEEL_REGEX, "");
}
function sanitize(str) {
  str = str.replace(ALIF_REGEX, "\u0627").replace(YA_REGEX, "\u064A").replace(TA_REGEX, "\u0647");
  str = removeTatwel(str);
  str = removeTashkel(str);
  return str;
}
export {
  ALIF_REGEX,
  TASHKEEL,
  TASHKEEL_REGEX,
  TATWEEL_REGEX,
  TA_REGEX,
  YA_REGEX,
  hasArabic,
  howArabic,
  howNotArabic,
  isArabic,
  removeArabic,
  removeNonArabic,
  removeTashkel,
  removeTatwel,
  sanitize
};
